### ISSUE TITLE
Capturing audio

### ISSUE THREAD

**Issue by aleclarson** [sebpiq/node-web-audio-api#35]
Do you recommend anything for capturing audio from a microphone with Node?

Or should something like [node-microphone](https://github.com/vincentsaluzzo/node-microphone) work fine?

Thanks.

**Comment by sebpiq** [comment]
Hey! I haven't tried recording audio, so sorry but I can't recommend anything.

**Comment by ruswerner** [comment]
@aleclarson Did you get `node-microphone` to work?

**Comment by dbkaplun** [comment]
+1 I was able to use https://github.com/Streampunk/naudiodon to record some audio, haven't done much else though.

**Comment by k2d222** [comment]
@dy 
It doesn't answer how to capture a microphone audio. In a browser, we would do `navigator.mediaDevices.getUserMedia({ audio: true })` to request access to the microphone as a `MediaStream`, and then create a source node with `ctx.createMediaStreamSource(mediaStream)`.

`getUserMedia` is beyond the scope of web-audio-api, but I couldn't find any node.js pollyfill for it, nor any node package compatible with web-audio-api that would provide a `MediaStream` capturing microphone audio.

**Comment by dy** [comment]
Right. [audio-mic](https://github.com/audiojs/audio-mic) is for that purpose.
Added example in readme.
Thanks for pointing out @k2d222, hope it works for you.

**Comment by k2d222** [comment]
On a 2nd thought, while your polyfill of `mediaDevices` is very convenient, I would recommend moving it to a separate package. It is technically not part of WebAudio, and its current implementation is quite janky, and does not match the library claim to be 100% CTS-compliant.

**Comment by k2d222** [comment]
The options you pass to `new MediaStreamAudioSourceNode(ctx, { numberOfChannels: 1, bitDepth: 16 })` are out of spec and should be at least documented in "Node extensions" in readme. Same for `src.pushData()`. Actually I don't understand this approach. IMO It would make more sense to provide a method to create a custom `MediaStream` instead of hacking into `MediaStreamAudioSourceNode`.

Also in readme: "See the [mic FAQ](https://github.com/audiojs/web-audio-api#how-do-i-capture-audio-from-the-microphone)" -> the link is broken.

**Comment by dy** [comment]
@k2d222 please have a look at https://github.com/audiojs/web-audio-api/pull/101/changes - would that be better?
I was considering `createMediaStream` method, but decided that escape hatch similar to `AudioContext({sinkId: stream})` would be more consistent. Maybe you're right.

**Comment by k2d222** [comment]
commented on the PR.
